### DISCUSSION TITLE
Flutter web apps supported?

### DISCUSSION THREAD

**Discussion by pandoras-toolbox** [mobile-dev-inc/Maestro#1240] [Q&A]
Are only Flutter mobile apps supported by Maestro or also Flutter web apps?

And do you plan to support Flutter web apps?

Currently we can rule out Maestro because it cannot be fully self-hosted, but if that changes, then it would be good to know if Maestro supports or will support testing Flutter web apps.

**Comment by portyu9** [comment]
Hi @pandoras-toolbox 

Yes — this has changed since the original question. **Flutter Web is now supported by Maestro.**

Maestro relies on Flutter's Semantics information, so make interactive elements addressable using `Semantics`, `semanticLabel`, or a semantics `identifier`.

For example:

```dart
Semantics(
  identifier: 'login_button',
  child: ElevatedButton(
    onPressed: login,
    child: const Text('Sign In'),
  ),
)
```

and then:

```yaml
url: https://example.com
---
- launchApp
- tapOn:
    id: "login_button"
```

Flutter Web testing follows Maestro's normal web-testing path. Web/browser support is currently based on managed Chromium, so keep that browser limitation in mind.

Regarding self-hosting: the **Maestro CLI executes tests locally** against your own browser/emulator/device, so you do not need Maestro Cloud for test execution. Maestro Cloud is the hosted service for scalable/parallel managed execution and reporting; it is not required just to run Flutter Web tests.

Please mark this answer as accepted if it helped, thank you.
